internal/factsengine/gatherers/sysctl_test.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/trento-project/agent/test/helpers"
+)
+
+type SysctlTestSuite struct {
+	suite.Suite
+	mockExecutor *utilsMocks.CommandExecutor
+}
+
+func TestSysctlTestSuite(t *testing.T) {
+	suite.Run(t, new(SysctlTestSuite))
+}
+
+func (suite *SysctlTestSuite) SetupTest() {
+	suite.mockExecutor = new(utilsMocks.CommandExecutor)
+}
+
+func (suite *SysctlTestSuite) TestSysctlGathererNoArgumentProvided() {
+	mockOutputFile, _ := os.Open(helpers.GetFixturePath("gatherers/sysctl.output"))
+	mockOutput, _ := io.ReadAll(mockOutputFile)
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(mockOutput, nil)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "no_argument_fact",
+			Gatherer: "sysctl",
+		},
+		{
+			Name:     "empty_argument_fact",
+			Gatherer: "sysctl",
+			Argument: "",
+		},
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:  "no_argument_fact",
+			Value: nil,
+			Error: &entities.FactGatheringError{
+				Message: "missing required argument",
+				Type:    "sysctl-missing-argument",
+			},
+		},
+		{
+			Name:  "empty_argument_fact",
+			Value: nil,
+			Error: &entities.FactGatheringError{
+				Message: "missing required argument",
+				Type:    "sysctl-missing-argument",
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlGathererNonExistingKey() {
+	mockOutputFile, _ := os.Open(helpers.GetFixturePath("gatherers/sysctl.output"))
+	mockOutput, _ := io.ReadAll(mockOutputFile)
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(mockOutput, nil)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "madeup_fact",
+			Gatherer: "sysctl",
+			Argument: "madeup.fact",
+		},
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:  "madeup_fact",
+			Value: nil,
+			Error: &entities.FactGatheringError{
+				Message: "requested value not found in sysctl output: madeup.fact",
+				Type:    "sysctl-value-not-found",
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlCommandNotFound() {
+	suite.mockExecutor.On("Exec", "sysctl", "-a").Return(nil, exec.ErrNotFound)
+
+	c := gatherers.NewSysctlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "fs.inotify.max_user_watches",
+			Gatherer: "sysctl",
+			Argument: "fs.inotify.max_user_watches",
+		},
+	}
+
+	expectedError := &entities.FactGatheringError{
+		Message: "error executing sysctl command: executable file not found in $PATH",
+		Type:    "sysctl-cmd-error",
+	}
+
+	factResults, err := c.Gather(factRequests)
+
+	suite.EqualError(err, expectedError.Error())
+
+	suite.Empty(factResults)
+}
+
+func (suite *SysctlTestSuite) TestSysctlGatherer() {
```

Comment:
As we are going with the "map" version, could we add a more complex scenario?
Like asking for `user` and getting all the children values

Reply:
You could've added the request in this very same test, but well, a new test is always ok